test/unit/formatting_stack/util/diff.clj

```diff
@@ -0,0 +1,78 @@
+(ns unit.formatting-stack.util.diff
+  (:require
+   [clojure.test :refer [are deftest is testing use-fixtures]]
+   [formatting-stack.test-helpers :refer [with-mocked-diff-path]]
+   [formatting-stack.util.diff :as sut]))
+
+(use-fixtures :once with-mocked-diff-path)
+
+(deftest diff->line-numbers
```

Comment:
checking out the files 1.patch to 4.patch, I think we could benefit from extra coverage via:

* diffs where there's /dev/null in either left or right hand
* diffs denoting a file renaming